### DISCUSSION TITLE
值与默认值的选择合并策略感觉有点问题啊

### DISCUSSION THREAD

**Discussion by liuweiGL** [alibaba/formily#1317] [Q&A 2.x] [answered]
## 文档
https://core.formilyjs.org/guide/form#%E6%95%B0%E6%8D%AE%E6%A8%A1%E5%9E%8B

## CodeSandBox
https://codesandbox.io/s/jolly-lumiere-f5zy2?file=/App.tsx

## 问题
在 ArrayTable 场景中，肯定是 `form` 对象先有值，然后再去渲染 `Field` 组件，这就导致 schema 中设置的 `default` 值一定优先于 form 表单设置的值。所以，这个行为应该不对吧，还是我的用法是错误的呢？

**Comment by liuweiGL** [comment]
我记得之前是表单中改字段没有值的时候会使用 default 值，而不是这样直接覆盖。

**Comment by janryWang** [comment]
这个场景有问题的，values的优先级会比intialValue优先级要高，我单测没覆盖到

**Comment by liuweiGL** [comment]
直接覆盖的会导致一个问题：
我有一个 ArrayTable，然后 a 字段是设置 default 为 1 的。
1. 用户点击添加按钮：新增的行使用 default 值行为是对的。
2. 用户点击选择弹框（自定义组件），然后从弹框中勾选 3 行数据，点击确定，ArrayTable 就有三行，但是这三行还是用 default 值就不对了。

**Comment by liuweiGL** [comment]
我现在没办法解决这个问题呢，有好的建议吗？

**Comment by janryWang** [comment]
这个算是个Bug吧，我得修复一下，等下个版本

**Comment by janryWang** [comment, marked as answer]
已经修复了，马上发版

**Reply by liuweiGL** [reply]
这么快，666
